provider/docker/config.go

```
@@ -322,22 +311,49 @@ func getPort(container dockerData) string {
 	return ""
 }
 
+func (p *Provider) getPortBinding(container dockerData) (nat.PortBinding, error) {
+	port := getPort(container)
+	for netPort, portBindings := range container.NetworkSettings.Ports {
+		if strings.EqualFold(string(netPort), port+"/TCP") || strings.EqualFold(string(netPort), port+"/UDP") {
+			for _, p := range portBindings {
+				return p, nil
+			}
+		}
+	}
+
+	return nat.PortBinding{HostIP: "", HostPort: ""}, fmt.Errorf("Unable to find the external IP:Port for the container %q", container.Name)
+}
+
 func (p *Provider) getServers(containers []dockerData) map[string]types.Server {
 	var servers map[string]types.Server
 
 	for _, container := range containers {
-		ip := p.getIPAddress(container)
-		if len(ip) == 0 {
-			log.Warnf("Unable to find the IP address for the container %q: the server is ignored.", container.Name)
-			continue
+		var ip, port string
+
+		if p.UseBindPortIP {
+			portBinding, err := p.getPortBinding(container)
+			if err != nil {
+				log.Warnf("Unable to find a binding for the container %q: ignoring server", container.Name)
+				continue
+			}
+			if portBinding.HostIP != "0.0.0.0" {
```

fail fast: could you invert the condition:
```go
if portBinding.HostIP == "0.0.0.0" {
// ...
}
// ...
```
The `else` is not needed.